docs/ble.md

```diff
@@ -58,157 +72,171 @@ dictionary CharacteristicInit {
     NotifyCallback onNotify;            // optional
 };
 
-callback ReadCallback = void (unsigned long offset, FulfillReadCallback);
+interface Characteristic {
+    attribute ReadCallback onReadRequest;
+    attribute WriteCallback onWriteRequest;
+    attribute SubscribeCallback onSubscribe;
+    attribute UnsubscribeCallback onUnsubscribe;
+    attribute NotifyCallback onNotify;
+    attribute CharacteristicResult response;
+};
+
+callback ReadCallback = void (unsigned long offset,
+                              FulfillReadCallback fulfillReadCallback);
 callback WriteCallback = void (Buffer data, unsigned long offset,
-                               boolean withoutResponse, FulfillWriteCallback);
+                               boolean withoutResponse,
+                               FulfillWriteCallback fulfillWriteCallback);
 callback SubscribeCallback = void (unsigned long maxValueSize,
-                                   FulfillSubscribeCallback);
+                                   FulfillSubscribeCallback fullfillSubscribeCallback);
 callback FulfillReadCallback = void (CharacteristicResult result, Buffer data);
 callback FulfillWriteCallback = void (CharacteristicResult result);
 callback FulfillSubscribeCallback = void (Buffer data);
 
+enum CharacteristicResult { "RESULT_SUCCESS", "RESULT_INVALID_OFFSET", "RESULT_INVALID_ATTRIBUTE_LENGTH", "RESULT_UNLIKELY_ERROR" } ;
```

Comment:
When enumeration types are for strings (preferred today) then they should all be LOWER CASE and with HYPHEN between.. not underscore.

`enum CharacteristicResult { "success", "invalid-offset"... }`